Sources/Datadog/FeaturesIntegration/LoggingWithActiveSpanIntegration.swift

```diff
@@ -16,17 +16,22 @@ internal struct LoggingWithActiveSpanIntegration {
     /// Produces `Log` attributes describing the current active span.
     /// Returns `nil` and prints warning if global `Tracer` is not registered.
     var activeSpanAttributes: [String: Encodable]? {
-        guard let tracer = Global.sharedTracer as? Tracer else {
-            userLogger.warn("Tracing feature is enabled, but no `Tracer` is registered. The Tracing integration with Logging will not work.")
-            return nil
-        }
-
-        if let activeSpanContext = tracer.activeSpan?.context as? DDSpanContext {
+        let tracer = Global.sharedTracer as? Tracer
+        if let tracer = tracer,
+            let activeSpanContext = tracer.activeSpan?.context as? DDSpanContext {
             return [
                 Attributes.traceID: "\(activeSpanContext.traceID.rawValue)",
                 Attributes.spanID: "\(activeSpanContext.spanID.rawValue)"
             ]
+        } else if let environmentContext = Global.environmentContext {
+            return [
+                Attributes.traceID: "\(environmentContext.traceID.rawValue)",
+                Attributes.spanID: "\(environmentContext.spanID.rawValue)"
+            ]
         } else {
+            if tracer == nil {
+                userLogger.warn("Tracing feature is enabled, but no `Tracer` is registered. The Tracing integration with Logging will not work.")
+            }
```

Comment:
I wouldn't mix the ENV span context with `LoggingWithActiveSpanIntegration` as both mean something conceptually different. The first one is immutable (never changes) span context passed from external process, while the latter is a mutable active span generated within the process (it does change).

I think it would be way easier and more maintainable if we just expose two separate integrations (like we did for `LoggingWithRUMContextIntegration` and `TracingWithRUMContextIntegration`):
```swift
struct TracingWithEnvironmentSpanIntegration {
   /// Span context to be attached to all root spans in order to continue the trace
   /// injected by external process through ENV variables.
   let spanContext: (spanID: TracingUUID, traceID: TracingUUID)

   init?() {
      // return an instance if ENV parameters are available
      // return `nil` otherwise
   }
}

struct LoggingWithEnvironmentSpanIntegration {
   /// Additional log attributes describing the ENV span injected by external process.
   /// Adding those attributes to `Log` will correlate them with that span.
   let environmentSpanAttributes: [String: Encodable]

   init?() {
      // return an instance if ENV parameters are available
      // return `nil` otherwise
   }
}
```

Both in `Logger` and `Tracer` this can be laid off just like any other integration:
```swift
// Logger

/// Integration with Span context injected by environment. `nil` if not available.
internal let environmentSpanIntegration = LoggingWithEnvironmentSpanIntegration()

// Usage:
if let activeSpanAttributes = activeSpanIntegration?.activeSpanAttributes {
   combinedInternalAttributes.merge(activeSpanAttributes) { $1 }
} else if let environmentSpanIntegration = environmentSpanIntegration?.environmentSpanAttributes {
   combinedInternalAttributes.merge(environmentSpanAttributes) { $1 }
}
```
```swift
// Tracer

/// Integration with Span context injected by environment. `nil` if not available.
internal let environmentSpanIntegration = TracingWithEnvironmentSpanIntegration()

// Usage:
traceID: parentSpanContext?.traceID ?? (environmentSpanIntegration?.spanContext.traceID ?? tracingUUIDGenerator.generateUnique()),
spanID: tracingUUIDGenerator.generateUnique(),
parentSpanID: parentSpanContext?.spanID ?? environmentSpanIntegration?.spanContext.spanID,
```

IMO this would be more streamlined implementation, which isolates the entire thing to separate units and describes each place very well with comments. Also, it keeps equal level of abstraction in both `Logger` and `Tracer`, as both just install the integration and capture parameters from it.

We can also get rid of `Global. environmentContext`, if we capture it inside each integration (or even reuse one integration within another or decouple common thing to a third integration as we did in `RUMContextIntegration`).